pkg/apis/work/v1alpha2/binding_types_helper_test.go

```diff
@@ -129,3 +131,142 @@ func TestResourceBindingSpec_RemoveCluster(t *testing.T) {
 		})
 	}
 }
+
+func TestResourceBindingSpec_GracefulEvictCluster(t *testing.T) {
+	tests := []struct {
+		Name       string
+		InputSpec  ResourceBindingSpec
+		EvictEvent GracefulEvictionTask
+		ExpectSpec ResourceBindingSpec
+	}{
+		{
+			Name: "cluster not exist should do nothing",
+			InputSpec: ResourceBindingSpec{
+				Clusters: []TargetCluster{{Name: "m1"}, {Name: "m2"}, {Name: "m3"}},
+			},
+			EvictEvent: GracefulEvictionTask{FromCluster: "non-exist"},
+			ExpectSpec: ResourceBindingSpec{
+				Clusters: []TargetCluster{{Name: "m1"}, {Name: "m2"}, {Name: "m3"}},
+			},
+		},
+		{
+			Name: "evict cluster from head",
+			InputSpec: ResourceBindingSpec{
+				Clusters: []TargetCluster{{Name: "m1", Replicas: 1}, {Name: "m2", Replicas: 2}, {Name: "m3", Replicas: 3}},
+			},
+			EvictEvent: GracefulEvictionTask{
+				FromCluster: "m1", Replicas: pointer.Int32Ptr(1),
```

Comment:
`Replicas: pointer.Int32Ptr(1)` should not be needed.

Reply:
right.